### PR TITLE
Docker image generation

### PR DESCRIPTION
**Pull request by Finomnis**
This branch will create a docker image with hpx installed on it.
This could be used for rapid prototyping, external unit tests, writing minimal examples that reproduce errors, potential deployment in the cloud, continuous integration for external projects (like hpxcl), etc.

### CODE REVIEW

**Comment by Finomnis** [comment]
I will still need to add documentation, but this branch should be merged with master before we modify circle.yml in any way.

File: tools/docker/docker-compile.pl [hunk 1]
```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env perl
+
+# copyright: Maciej Pasternacki
+# http://3ofcoins.net/2013/09/22/flat-docker-images/
+#
+# TODO: replace with something homemade
+
```
Comment (hkaiser) [inline tools/docker/docker-compile.pl]: Please add the Boost License header

Comment (Finomnis) [inline tools/docker/docker-compile.pl]: U sure? We don't even have the copyright for this file, I pulled it in from that dude. He published it via gist without copyright note

Comment (hkaiser) [inline tools/docker/docker-compile.pl]: He added his copyright but published it without a license (which means it's in the public domain). That entitles us to relicense it under the Boost license, I don't see a problem with that.

Comment (Finomnis) [inline tools/docker/docker-compile.pl]: Well, technically he didn't even add his name in, I did that ;)
Should I strip his name out? Or is it polite to leave that in

Comment (hkaiser) [inline tools/docker/docker-compile.pl]: I'd leave his copyright in place, credit where credit is due...

**Comment by hkaiser** [comment]
LGTM, please go ahead and merge.

**Pull request merged**